### DISCUSSION TITLE
Connections: Integrate Toolbx and Distrobox environments

### DISCUSSION THREAD

**Discussion by tajirelimunuru** [wavetermdev/waveterm#775] [Ideas]
![image](https://github.com/user-attachments/assets/01cdbbbd-d5e7-44d7-b010-faa79015d2d7)

**Background**

One of the recent development and admin workflows is the use of toolboxes (OS container images). These provide isolated environments for development, testing, and running applications without affecting the host system. Projects like [Toolbx](https://containertoolbx.org/) and [Distrobox](https://distrobox.it/) have become widely used in this regard.

I know of  [Pytxis](https://gitlab.gnome.org/chergert/ptyxis) and at least one or two other terminal emulators that have integrated use of toolboxes _(I've been using [Ptyxis](https://flathub.org/apps/app.devsuite.Ptyxis) personally and have found the integration of container environments to be quite useful)_.

**Proposal for Wave**

My proposal is the integration of Toolbx and Distrobox environments within Wave. This feature could potentially be implemented as a new section under "Connections," alongside the existing options for remote machines.

To begin, it could just be the ability to use and switch between the boxes easily as currently with SHH connections. Later, there could be a create, start, stop, delete within the interface _(but that would be a bonus)_.

I'm excited to hear the community's thoughts on this idea.

**Comment by agoodfellow123** [comment]
I'm actually surprised wave doesn't have this since it has a similar feature, WSL. Which is basically Microsoft™ Distrobox. I would love to have this as well. For now though, I wonder if it's possible to hack something in using Docker since Distrobox is basically a Docker (if you chose docker to use it with) wrapper.

**Reply by esimkowitz** [reply]
Definitely in the pipeline, we prioritized WSL because we wanted to offer the best out-of-the-box experience for Windows users and our feedback showed it had a higher user base. Docker/OCI is definitely in the pipeline, though I can't share details on timelines yet. We'll work to try and align that implementation so it's compatible with Distrobox and Toolbx.
